### DISCUSSION TITLE
How get properties for a plugin?

### DISCUSSION THREAD

**Discussion by charles-niaid** [nikhilsbhat/terraform-provider-gocd#8] [Q&A] [answered]
Hello,

If I want to use [gocd_cluster_profile](https://registry.terraform.io/providers/nikhilsbhat/gocd/latest/docs/resources/cluster_profile) with the [ecs elastic agent plugin](https://github.com/gocd/gocd-ecs-elastic-agent/tree/master ), how can I get a list of available/valid property keys?

Thanks,
Charles

**Comment by nikhilsbhat** [comment, marked as answer]
@charles-niaid, the [GoCD API](https://api.gocd.org/current/#get-plugin-info) can provide all the properties of a given plugin.
However, none of the existing tools currently support fetching this information. 
I'll take this feedback into account and work on implementing a feature on both [data_source](https://registry.terraform.io/providers/nikhilsbhat/gocd/latest/docs/data-sources/plugin_info) and [gocd-cli](https://github.com/nikhilsbhat/gocd-cli) to retrieve all necessary plugin properties.

**Comment by nikhilsbhat** [comment]
Support for retrieving plugin properties has been added to the latest versions of both the [gocd-cli](https://github.com/nikhilsbhat/gocd-cli) and the Terraform provider's [data_source](https://registry.terraform.io/providers/nikhilsbhat/gocd/latest/docs/data-sources/plugin_info).
